internal/controller/postgrescluster/pgmonitor_test.go

```diff
@@ -578,3 +599,121 @@ func TestReconcilePGMonitorSecret(t *testing.T) {
+// When exporter is enabled with custom tls configureExporterTLS should add volumes,
+// volumeMounts, and an envVar to the template. Ensure that existing template configurations
+// are still present
+func TestConfigreExporterTLS(t *testing.T) {
+	// Define an existing template with values that could be overwritten
+	baseTemplate := &corev1.PodTemplateSpec{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name: naming.ContainerPGMonitorExporter,
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      "existing-volume",
+					MountPath: "some-path",
+				}},
+				Env: []corev1.EnvVar{{
+					Name:  "existing-env",
+					Value: "existing-value",
+				}},
+			}},
+			Volumes: []corev1.Volume{{
+				Name: "existing-volume",
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			}},
+		},
+	}
+
+	t.Run("Exporter disabled", func(t *testing.T) {
+		cluster := &v1beta1.PostgresCluster{}
+		template := baseTemplate.DeepCopy()
+		configureExporterTLS(cluster, template, nil)
+		// Template shouldn't have changed
+		assert.DeepEqual(t, template, baseTemplate)
+	})
+
+	t.Run("Exporter enabled no tls", func(t *testing.T) {
+		cluster := &v1beta1.PostgresCluster{
+			Spec: v1beta1.PostgresClusterSpec{
+				Monitoring: &v1beta1.MonitoringSpec{
+					PGMonitor: &v1beta1.PGMonitorSpec{
+						Exporter: &v1beta1.ExporterSpec{},
+					},
+				},
+			},
+		}
+		template := baseTemplate.DeepCopy()
+		configureExporterTLS(cluster, template, nil)
+		// Template shouldn't have changed
+		assert.DeepEqual(t, template, baseTemplate)
+	})
+
+	t.Run("Custom TLS provided", func(t *testing.T) {
+		cluster := &v1beta1.PostgresCluster{
+			ObjectMeta: metav1.ObjectMeta{Name: "test"},
+			Spec: v1beta1.PostgresClusterSpec{
+				Monitoring: &v1beta1.MonitoringSpec{
+					PGMonitor: &v1beta1.PGMonitorSpec{
+						Exporter: &v1beta1.ExporterSpec{
+							CustomTLSSecret: &corev1.SecretProjection{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "custom-exporter-certs",
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		template := baseTemplate.DeepCopy()
+
+		testConfigMap := &corev1.ConfigMap{
+			ObjectMeta: naming.ExporterWebConfigMap(cluster),
+		}
+
+		// What happens if the template already includes volumes/Mounts and envs?
+		configureExporterTLS(cluster, template, testConfigMap)
+
+		// Did we configure the cert volume and the web config volume while leaving
+		// existing volumes in place?
+		assert.Assert(t, marshalMatches(template.Spec.Volumes, `
+- emptyDir: {}
+  name: existing-volume
+- name: exporter-certs
+  projected:
+    sources:
+    - secret:
+        name: custom-exporter-certs
+- configMap:
+    name: test-exporter-web-config
+  name: web-config
+		`))
+
+		// Is the exporter container in position 0?
+		assert.Assert(t, template.Spec.Containers[0].Name == naming.ContainerPGMonitorExporter)
```

Comment:
:question: Are we just verifying position here before running the subsequent tests or is there another reason for this check?

Reply:
Yup, just checking the position.